modules/core/client/app/init.js

```diff
@@ -36,11 +36,13 @@ angular.module(ApplicationConfiguration.applicationModuleName).run(function ($ro
         }
       }
     }
+
   });
 
   // Record previous state
   $rootScope.$on('$stateChangeSuccess', function (event, toState, toParams, fromState, fromParams) {
     storePreviousState(fromState, fromParams);
+    $rootScope.$state = $state;
```

Comment:
http://stackoverflow.com/a/23814161/2228227 as @trainerbill suggested would be better, IMO